ModAssistant/Pages/Mods.xaml.cs

```diff
@@ -390,22 +393,44 @@ private async Task InstallMod(Mod mod, string directory)
                 return;
             }
 
-            using (Stream stream = await DownloadMod(Utils.Constants.BeatModsURL + downloadLink))
-            using (ZipArchive archive = new ZipArchive(stream))
+            List<ZipArchiveEntry> files;
+
+            do
             {
-                foreach (ZipArchiveEntry file in archive.Entries)
+                files = new List<ZipArchiveEntry>(filesCount);
+
+                using (Stream stream = await DownloadMod(Utils.Constants.BeatModsURL + downloadLink))
+                using (ZipArchive archive = new ZipArchive(stream))
                 {
-                    string fileDirectory = Path.GetDirectoryName(Path.Combine(directory, file.FullName));
-                    if (!Directory.Exists(fileDirectory))
+                    foreach (ZipArchiveEntry file in archive.Entries)
                     {
-                        Directory.CreateDirectory(fileDirectory);
-                    }
+                        string fileDirectory = Path.GetDirectoryName(Path.Combine(directory, file.FullName));
+                        if (!Directory.Exists(fileDirectory))
+                        {
+                            Directory.CreateDirectory(fileDirectory);
+                        }
 
-                    if (!string.IsNullOrEmpty(file.Name))
-                    {
-                        await ExtractFile(file, Path.Combine(directory, file.FullName), 3.0, mod.name, 10);
+                        if (!string.IsNullOrEmpty(file.Name))
+                        {
+                            foreach (Mod.DownloadLink download in mod.downloads)
+                            {
+                                foreach (Mod.FileHashes fileHash in download.hashMd5)
+                                {
+                                    if (fileHash.hash == Utils.CalculateMD5FromStream(file.Open()))
```

Comment:
Shouldn't `file.Open()` be wrapped with a `using`?

Reply:
Right, I forgot that.

Reply:
Can you fix that? I have to leave now, I'll come a bit later.

Reply:
Nevermind, done in the next commit.